_posts/2020-01-16-spring-bean-life-cycle.md

```diff
@@ -0,0 +1,161 @@
+---
+title: "The Lifecycle of Spring Beans"
+categories: [spring-boot]
+modified: 2020-01-16
+excerpt: "An in-depth look at the paging support provided by Spring Data for querying
+          Spring Web MVC controllers and Spring Data repositories."
+image:
+  auto: 0012-pages
+
+---
+
+## What are Spring Beans?
+ > Spring Beans are Java Objects or Instances which is created and managed by Spring container.
+
+Spring `@ApplicationContext` is responsible for creating a Spring bean and manage the 
+Spring bean life cycle.
+
+Below, We have created a simple Spring bean for `Foo` class using `@Component` annotation.
+
+```java
+@Component
+Class Foo { //variables and methods ...}
+```
+## The Spring Bean Lifecycle
+
+The Bean's life cycle consists of 2 phases `Post-initialization` and `Pre-destruction`. 
+
+Below are the Phase 1, `Post-initialization` steps while creating a Bean.
+1. Bean Creation → using Annotations or XML Bean Configs.
+2. Bean Instantiate → In this step Bean will be loaded in `JVM` memory.
+3. Populating Bean properties →  Container will Create a Bean `id`, `scope`, `default values`.
+4. Implementing `BeanNameAware setBeanName()` → Getting to know the Bean Name created by Spring container.
+5. Implementing `BeanClassLoaderAware setBeanClassLoder()` → Getting to know the class is loaded in Spring container Bean Factory.
+6. Implementing `BeanFactoryAware setBeanFactory()` → Getting to know the application Bean and its dependencies. 
+7. Implementing `ApplicationContextAware setApplicationContextAware()` → Getting to know the total number of beans created in the application and etc…
+8. `@PostConstruct` → will be called the annotated method.
+9. Implementing `InitializingBean afterPropertiesSet()` → method will be called after setting the bean properties.
+10. `@Bean init-method` → called specified method inside an `init-method` attribute.
+11. Bean is Ready to use.
+
+Below is Phase 2, `Pre-destruction` steps while destroying a Bean.
+
+1. `@PreDestroy` → Annotated method will be called.
+2. `DisposiableBean destroy()` → will be called at the closing.
+3. `@Bean destroy-method` → called specified method inside the `destroy-method` attribute.
+4. Bean will be destroyed.
+
+## Why Would I Need to Hook into the Bean Lifecycle?

```

Comment:
The examples you listed are great! This is what makes the article different from all the others about the Spring Bean lifecycle out there! But we lose a lot of potential if we only list them with a single sentence (also I don't understand some of them by just reading the one sentence).

How about moving this section to after the "Hooking into ..." section. Then, expand each bullet point in the list to its own subsection where you can refer to the solutions in the "Hooking into..." section to explain why and how you would implement each use case. Doesn't need to be a lot of text, but a couple sentences for each use case would be great.